### ISSUE TITLE
Failure when CPM_SOURCE_CACHE is specified with CLion

### ISSUE THREAD

**Issue by Banderi** [cpm-cmake/CPM.cmake#371]
I'm using the latest version of CPM, freshly downloaded. Using CLion, CMake 3.24/3.22 and Windows 10.
If `CPM_SOURCE_CACHE` is set, it "fails" and the script configuration crashes, complaining that the folders already exist. (They do not. This is from a completely clean folder with no cache directories.)
This is the CMake script I have:
```cmake
set(CPM_SOURCE_CACHE .cache/CPM)
CPMAddPackage(
    GIT_TAG release-2.0.22
    GITHUB_REPOSITORY libsdl-org/SDL
)
```
And this is the output:
```
CPM_SOURCE_CACHE: .cache/CPM
-- CPM: adding package SDL@ (release-2.0.22 to E:/Git/Antimony-III/.cache/CPM/sdl/22376c56c7faadb422628d46bf5157d6eb41d577)
[1/9] Creating directories for 'sdl-populate'
[1/9] Performing download step (git clone) for 'sdl-populate'
fatal: destination path '22376c56c7faadb422628d46bf5157d6eb41d577' already exists and is not an empty directory.
fatal: destination path '22376c56c7faadb422628d46bf5157d6eb41d577' already exists and is not an empty directory.
fatal: destination path '22376c56c7faadb422628d46bf5157d6eb41d577' already exists and is not an empty directory.
-- Had to git clone more than once:
          3 times.
CMake Error at E:/Git/Antimony-III/cmake-build-release/_deps/sdl-subbuild/sdl-populate-prefix/tmp/sdl-populate-gitclone.cmake:31 (message):
  Failed to clone repository: 'https://github.com/libsdl-org/SDL.git'
```

Upon reloading the CMake config, it also either "fails" to remove the previous cache folders -
```
Error removing directory "E:/Git/Antimony-III/.cache/CPM/sdl2/22376c56c7faadb422628d46bf5157d6eb41d577".
CMake Error at E:/Git/Antimony-III/cmake-build-release/_deps/sdl2-subbuild/sdl2-populate-prefix/tmp/sdl2-populate-gitclone.cmake:12 (message):
  Failed to remove directory:
  'E:/Git/Antimony-III/.cache/CPM/sdl2/22376c56c7faadb422628d46bf5157d6eb41d577'
```
- or complains that the cache is dirty:
```
CMake Warning at cmake/CPM.cmake:679 (message):
  Cache for SDL2
  (E:/Git/Antimony-III/.cache/CPM/sdl2/22376c56c7faadb422628d46bf5157d6eb41d577)
  is dirty
```

What is bizarre is that the git command actually _does_ run in the background silently and CLion reports the project's CMake config as "loading" until the repo is downloaded - but because of the error, the CMake parser actually stops midway and corrupts the build cache. If I wait until it finished downloading and try to build the project, it will complain that the project cache is incomplete.
```
C:\Users\Banderi\AppData\Local\JetBrains\Toolbox\apps\CLion\ch-0\221.5591.52\bin\cmake\win\bin\cmake.exe --build E:\Git\Antimony-III\cmake-build-debug --target Antimony3 -j 9
Error: could not find CMAKE_PROJECT_NAME in Cache
```

If I run CMake manually with CLI with my project, it creates all the files and compiles just fine.
Running from CLion without `CPM_SOURCE_CACHE` set also runs and compiles fine.

**Comment by Banderi** [comment]
Found the culprit.

When multiple target configurations are present in CLion (e.g. Debug, Release, etc.) the IDE runs CMake configurations in *parallel* for each build type -- this effectively causes the process to succeed the first time and fail for the others.

A temporary workaround is to set `CPM_SOURCE_CACHE` to something like `.cache/CPM-${CMAKE_BUILD_TYPE}` but this seems very wasteful and will effectively clone the repository twice.

**Comment by TheLartians** [comment]
Happy to hear you could solve the issue!

**Comment by PercentBoat4164** [comment]
I too have encountered this problem. Would it be possible to integrate this code into CPM to eliminate this issue in the future? Perhaps there are downsides to this that I am not seeing?

I was able to set up a block of code that looks like
```CMake
# Add all packages
file(LOCK ${CPM_SOURCE_CACHE} DIRECTORY GUARD FILE)
CPMAddPackage(NAME Ccache.cmake GITHUB_REPOSITORY TheLartians/Ccache.cmake VERSION 1.2)

CPMAddPackage(NAME AssImp GITHUB_REPOSITORY assimp/assimp VERSION 5.2.5)
CPMAddPackage("gh:g-truc/glm#master")
CPMAddPackage("gh:nigels-com/glew#glew-2.2.0")
CPMAddPackage("gh:glfw/glfw#3.3.8")
CPMAddPackage("gh:gabime/spdlog@1.11.0")
CPMAddPackage("gh:charles-lunarg/vk-bootstrap@0.6")
CPMAddPackage("gh:GPUOpen-LibrariesAndSDKs/VulkanMemoryAllocator@3.0.1")
file(LOCK ${CPM_SOURCE_CACHE} DIRECTORY GUARD FILE RELEASE)
```

This code worked well for me, but if the file locking was integrated into CPM, then the parallelization of the CMake scripts would not be as hindered as they are with the configuration above.

**Comment by TheLartians** [comment]
Is this fixed with #427?

**Comment by TheLartians** [comment]
Update: seems this is fixed according to https://github.com/cpm-cmake/CPM.cmake/issues/293#issuecomment-1407416655.
